### ISSUE TITLE
Try to get on the official WordPress oEmbed whitelist

### ISSUE THREAD

**Issue by reefdog** [documentcloud/wordpress-documentcloud#15]
See https://codex.wordpress.org/Embeds#Okay.2C_So_What_Sites_Can_I_Embed_From.3F

If we were whitelisted, then this plugin would be unnecessary (except to support older versions of WP).

So this issue is also the "obsolete this plugin" issue.

**Comment by eyeseast** [comment]
I think this is worth doing, but I don't think it would make the plugin obsolete. You might still want to pass in viewer configuration on a per-document basis.

We could also implement oembed-style embedding via this plugin, like Storify does.

**Comment by knowtheory** [comment]
@eyeseast we're implementing an oEmbed endpoint right now, and were planning on kicking around this wordpress plugin as our test consumer.

What's Storify do regarding oEmbed? :)

**Comment by reefdog** [comment]
@eyeseast That's totally true, we'd discussed that internally and I'd forgotten that. There's definitely still room for a more advanced plugin even after whitelisting.

**Comment by eyeseast** [comment]
See point 3 here: https://wordpress.org/plugins/storify/faq/. You just include a link on its own line, like so:

```
I'm writing this post here.

https://storify.com/eyeseast/christensenonrap

That storify was awesome.
```

The storify link will get turned into an embed code.

**Comment by aschweigert** [comment]
The storify plugin might be a pretty good reference actually if you're not familiar with how WP handles some of this stuff (or, is able to handle some of this stuff) from the theme/plugin side

**Comment by iparamonau** [comment]
Hi guys. Sorry to cut in, I monitor "oEmbed" topic here on GitHub. 

I am one of the guys behind Iframely and we'd be happy to be your test oEmbed consumer. And actually work with you as I am sure you will run into document height issues. 

What Storify does as embed code is that they give iframe of 750 height and also include javascript that adjusts the height onload through window.postMessage. https://github.com/davidjbradshaw/iframe-resizer - if you'd rather not implement it yourself. 

Here's to help you debug your oEmbed: http://iframely.com/debug/
Here's the simple reference for wordpress - https://github.com/itteco/iframely-wordpress 

We have a bunch of other tools so you could see your oEmbeds in a variety of ways. Let me know.

**Comment by reefdog** [comment]
Steps to getting here:

1. [ ] oEmbed service deployed to production, stable and tested
2. [ ] Resource pages properly discoverable (not necessary for inclusion, just a dotted i / crossed t)
3. [ ] Document how someone would use this sans plugin
4. [ ] Ask WordPress nicely?

**Comment by TheRoccoB** [comment]
Were you ever able to get this going? What was the process for getting whitelisted w/ wordpress? I may be implementing something similar, and would love to be whitelisted.

**Comment by knowtheory** [comment]
Hey @TheRoccoB, we haven't pursued whitelisting further with wordpress.com.  We do know that the plugin's gotten a review for Wordpress VIP, but that was handled at the level of our users asking their VIP teams to investigate.

**Comment by TheRoccoB** [comment]
Got it. Sounds like you have to be reasonably established for them to whitelist you. Thanks for letting me know!
